### PR TITLE
Improve drawing a polygon while also using zoom

### PR DESCRIPTION
**Pull request by soyfestivo**
Fixed a bug where while drawing a polygon if you zoomed in or out the shape was not translated to the new zoom level

zoom in while drawing polygon3

### CODE REVIEW

File: src/canvastools/ts/CanvasTools/Selection/Component/Element.ts [hunk 1]
```diff
@@ -70,8 +70,10 @@ export abstract class Element implements IHideable, IResizable {
      * Resizes element to specified `width` and `height`.
      * @param width - New element `width`.
      * @param height - New element `height`.
+     * @param oldWidth - New element `height`.
+     * @param height - New element `height`.
      */
-    public resize(width: number, height: number) {
+    public resize(width: number, height: number, oldWidth?: number, oldHeight?: number) {
```
Comment (samclee) [inline src/canvastools/ts/CanvasTools/Selection/Component/Element.ts:76]: What are these optional args needed for? I'm guessing it's somethingabout the PolygonSelector's "resize()" but I'm not totally sure

File: src/canvastools/ts/CanvasTools/CanvasTools.Editor.ts [hunk 2]
```diff
@@ -202,6 +202,18 @@ export class Editor {
...
+            iconFile: "polygon-selection.svg",
+            tooltip: "Polygon-selection (O)",
+            key: ["O", "o"],
+            actionCallback: (action, rm, sl) => {
+                sl.setSelectionMode({ mode: SelectionMode.POLYGON });
+                sl.show();
+            },
+            activate: false,
+        },
         {
```
Comment (pritamDas9191) [inline src/canvastools/ts/CanvasTools/CanvasTools.Editor.ts:217]: why do we need polygon select in RECT features ?

File: src/canvastools/ts/CanvasTools/Selection/Selectors/PolygonSelector.ts [hunk 3]
```diff
@@ -91,12 +91,21 @@ export class PolygonSelector extends Selector {
 
     /**
      * Resizes the selector to specified `width` and `height`.
-     * @param width - The new `width`.
-     * @param height - The new `height`.
+     * @param newWidth - The new `width`.
+     * @param newHeight - The new `height`.
```
Comment (pritamDas9191) [inline src/canvastools/ts/CanvasTools/Selection/Selectors/PolygonSelector.ts:95]: nit: add the optional param jsdoc too ?

File: src/canvastools/ts/CanvasTools/Selection/Component/Element.ts (hunk 1 above)
Comment (soyfestivo) [reply src/canvastools/ts/CanvasTools/Selection/Component/Element.ts:76]: We need to know by how much it's changing to scale down the `PolygonSelector` so we can update the coordinates proportionally to the change.

File: src/canvastools/ts/CanvasTools/CanvasTools.Editor.ts (hunk 2 above)
Comment (soyfestivo) [reply src/canvastools/ts/CanvasTools/CanvasTools.Editor.ts]: Right, I added this for testing. I'll remove.

**Pull request merged**